### PR TITLE
Add JUnit tests for Stats classes

### PR DESCRIPTION
**Pull request by Zhi-You**
Part of #105

### CODE REVIEW

File: src/test/java/seedu/eduke8/Eduke8Test.java [hunk 1]
```diff
@@ -28,11 +31,17 @@
...
+    protected static final int POINTS_FOR_CORRECT_ANSWER_WITHOUT_HINT_SHOWN = 2;
+    protected static final int POINTS_FOR_CORRECT_ANSWER_WITH_HINT_SHOWN = 1;
+    protected static final int POINTS_FOR_INCORRECT_ANSWER = 0;
+
     protected static final String DATA_TEST_INVALID_PATH = "data/test/invalid.json";
     protected static final String DATA_TEST_EMPTY_JSON = "data/test/empty.json";
     protected static final String DATA_TEST_WRONG_FORMAT_JSON = "data/test/wrong_format.json";
     protected static final String DATA_TEST_MISSING_KEY_JSON = "data/test/missing_key.json";
 
+    // Creates a TopicList object with two topics that have 3 questions each
```
Comment (kstonekuan) [inline src/test/java/seedu/eduke8/Eduke8Test.java:44]: Nice, makes it alot clearer

File: src/test/java/seedu/eduke8/hint/HintTest.java [hunk 2]
```diff
@@ -13,4 +14,13 @@ void getHintDescription_hintDescription_returnsHintDescription() {
 
         assertEquals(PLACEHOLDER_HINT_DESCRIPTION, hint.getDescription());
     }
+
+    @Test
+    void wasShown_hintShownToUser_expectsTrue() {
```
Comment (kstonekuan) [inline src/test/java/seedu/eduke8/hint/HintTest.java:19]: Nice improvement to code coverage

File: src/test/java/seedu/eduke8/stats/StatsCalculatorTest.java [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+import seedu.eduke8.common.Displayable;
+import seedu.eduke8.question.Question;
+import seedu.eduke8.question.QuestionList;
+import seedu.eduke8.topic.Topic;
+import seedu.eduke8.topic.TopicList;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+class StatsCalculatorTest extends Eduke8Test {
```
Comment (kstonekuan) [inline src/test/java/seedu/eduke8/stats/StatsCalculatorTest.java:14]: Good addition to the junits

**Review by kstonekuan — APPROVED** [review]
LGTM

File: src/test/java/seedu/eduke8/stats/StatsCalculatorTest.java [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+        TopicList testTopicList = createTestTopicList();
+
+        for (Displayable topic : testTopicList.getInnerList()) {
+            QuestionList topicQuestionList = ((Topic) topic).getQuestionList();
+            expectedQuestionCount += topicQuestionList.getCount();
+        }
+
+        assertEquals(expectedQuestionCount, statsCalculator.calculateTotalQuestionCount());
+    }
+}
```
Comment (josephhhhhhhhh) [inline src/test/java/seedu/eduke8/stats/StatsCalculatorTest.java:120]: Nice, this covers a lot of ground.

File: src/test/java/seedu/eduke8/question/QuizQuestionsManagerTest.java [hunk 5]
```diff
@@ -5,57 +5,48 @@
...
 
+    /*
+     * Tests for first parameter of QuizQuestionsManagerConstructor, while keeping second parameter valid.
+     */
 
-    // This test tests for getQuizQuestionsCount() method too
+    // Within acceptable range of questions, positive partition
     @Test
-    void quizQuestionsManagerConstructor_twoQuizQuestionsFromThreeTopicQuestions_returnsCountOfTwo()
+    void quizQuestionsManagerConstructor_validNumberOfQuestionsForQuiz_returnsCountOfThreeQuizQuestions()
```
Comment (anqi20) [inline src/test/java/seedu/eduke8/question/QuizQuestionsManagerTest.java:24]: This makes it a lot clearer!

File: src/test/java/seedu/eduke8/stats/TopicalStatsCalculatorTest.java [hunk 6]
```diff
@@ -0,0 +1,138 @@
...
+        QuestionList questionList = new QuestionList(questions);
+        Topic topic1 = new Topic(PLACEHOLDER_TOPIC_ONE_DESCRIPTION, questionList);
+
+        return topic1;
+    }
+
+    private int calculatePointsForTwoCorrectQuestionsOneHintUsed() {
+        return POINTS_FOR_CORRECT_ANSWER_WITHOUT_HINT_SHOWN + POINTS_FOR_CORRECT_ANSWER_WITH_HINT_SHOWN;
+    }
+}
```
Comment (josephhhhhhhhh) [inline src/test/java/seedu/eduke8/stats/TopicalStatsCalculatorTest.java:138]: Alot of coverage here as well, great implementation of the Junit test!

**Review by josephhhhhhhhh — APPROVED** [review]
Overall looks really good!

**Pull request merged**